src/Fantomas/IgnoreFile.fs

```diff
@@ -77,16 +77,7 @@ module IgnoreFile =
             let fullPath = fs.Path.GetFullPath(file)
 
             try
-                let path =
-                    if fullPath.StartsWith ignoreFile.Location.Directory.FullName then
-                        fullPath.[ignoreFile.Location.Directory.FullName.Length + 1 ..]
-                    else
-                        // This scenario is a bit unexpected - it suggests that we are
-                        // trying to ask an ignorefile whether a file that is outside
-                        // its dependency tree is ignored.
-                        fullPath
-
-                ignoreFile.IsIgnored path
+                ignoreFile.IsIgnored fullPath
```

Comment:
@Smaug123 I believe the problem of using a relative path here is the following:
In `loadIgnoreList` we process the path with
https://github.com/fsprojects/fantomas/blob/47c6ac6c15abf2830342866987ddd496be326c05/src/Fantomas/IgnoreFile.fs#L54-L58 

Looking at the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=net-7.0#remarks), they mention `GetFullPath` is invoked before calculating the difference.

In the case of the `cli tool`, the relative will be resolved with the `pwd` because that is where we assume the `ignore file` lives. For the daemon, on the other hand, the `pwd` could be something really random and thus the difference is calculated wrongly.

Does this make sense to you?

Reply:
Sigh, yeah, that does make sense. But might it be better to only pass fully-qualified paths?

Reply:
> But might it be better to only pass fully-qualified paths?

Something like [this](https://github.com/fsprojects/fantomas/pull/2656/commits/5ccb13a3d328127ed034140ffcd10abf322ea580)?